### PR TITLE
Additional CLI parameters to configure cluster apps and EC2 role

### PR DESCRIPTION
**Pull request by amajedi**
This pull request adds two (backwards-compatible) additional CLI parameters to configure the `jobflow_role` and set of `Applications` installed on the EMR cluster.

The release of this should bump the app to version `v1.1.0`.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,13 @@ Changelog
 Releases
 --------
 
+v1.1.0 (2019-07-13)
+~~~~~~~~~~~~~~~~~~~
+
+* Added `jobflow_role` CLI argument to configure cluster EC2 Instance Profile
```
Comment (ksindi) [inline CHANGELOG.rst]: Let's change it to present e.g. `Add `jobflow_role` CLI`

File: sparksteps/cluster.py [hunk 2]
```diff
@@ -54,6 +55,21 @@ def parse_conf(raw_conf_list):
...
+    Given a list of app name strings,
+    returns formatted application configuration value.
+
+    Examples:
+        >>> from pprint import pprint
+        >>> pprint(parse_apps(['hadoop', 'spark']))
+        [{'Name': 'Hadoop', 'Name': 'Spark'}]
+    """
+    return sorted(
+        [{'Name': app_name.capitalize()} for app_name in set(raw_app_list)],
```
Comment (ksindi) [inline sparksteps/cluster.py]: Interesting idea. I wonder if there's too much magic here where we autocapitalize.

File: sparksteps/defaults.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+"""
+Application CLI defaults
+"""
+DEFAULT_JOBFLOW_ROLE = 'EMR_EC2_DefaultRole'
+DEFAULT_APP_LIST = ['Hadoop', 'Spark']
```
Comment (ksindi) [inline sparksteps/defaults.py]: Should we just put these at the top of `cluster.py` and import from there? Not a huge fan of constants/defaults/utils type of files.

File: CHANGELOG.rst (hunk 1 above)
Comment (amajedi) [reply CHANGELOG.rst]: Also updated the rest of the changelog items to present tense

File: sparksteps/cluster.py (hunk 2 above)
Comment (amajedi) [reply sparksteps/cluster.py]: It should work as expected in all cases:

```python
'spark'.capitalize() == 'Spark'
'SPARK'.capitalize() == 'Spark'
'Spark'.capitalize() == 'Spark'
```

File: sparksteps/defaults.py (hunk 3 above)
Comment (amajedi) [reply sparksteps/defaults.py]: hmm, yeah a separate file might be overkill at the moment, was thinking of making it more future proof but can break it up into a separate file later when more constants are expressed. Sounds good.

**Pull request merged**